### DISCUSSION TITLE
SOC Accounts - Password Complexity

### DISCUSSION THREAD

**Discussion by xfaith** [Security-Onion-Solutions/securityonion#8072] [Unsupported Versions] [answered]
Has anyone tried or is it recommended to enforce better password complexity.  From reading the so-user it looks like it enforces 8 chars min, 72 max, 

Things like UpperCase/LowerCase,Numbers,SpecialCharacters, ect.    
Is there a password age length of SOC accounts?   

And  best policy if making updates to so-user w/ salts to keep the changes?

**Comment by dougburks** [comment, marked as answer]
> Has anyone tried or is it recommended to enforce better password complexity.

Rather than increasing password complexity, most organizations prefer to use multi-factor authentication:
https://docs.securityonion.net/en/2.3/mfa.html

**Reply by xfaith** [reply]
Doug, agree with MFA, I saw that currently SO can utilize Google Auth, any possiblilty of moving forward with non-phone related MFA, have not really looked into details on Kratos, so unsure of its full capabilities.

**Reply by dougburks** [reply]
I suspect you could find a TOTP app that doesn't require a phone. From https://docs.securityonion.net/en/2.3/mfa.html:
> Currently only Google Authenticator has been tested, however other authenticator apps that implement the time-based one-time password (TOTP) specification could also work.

**Comment by xfaith** [comment]
I see in Kratos 0.9 they have added USB-Keys (WebAuthn-supported authentication mechanisms).  Is there plans to implement these types of  logins for SecurityOnion in the future?

**Reply by dougburks** [reply]
We may look at this at some point in the future but no promises. For the time being, TOTP is the way to go.
